pandas/parser.pyx

```diff
@@ -714,7 +714,9 @@ cdef class TextReader:
                     start = self.parser.line_start[0]
 
                 # e.g., if header=3 and file only has 2 lines
-                elif self.parser.lines < hr + 1:
+                if (self.parser.lines < hr + 1
+                    and not isinstance(self.orig_header, list)) or (
```

Comment:
Counteract the extension to the header added in: https://github.com/pandas-dev/pandas/blob/master/pandas/parser.pyx#L519 The issue is that if the header is passed in as a list, it's extended to enable reading in the index name line (I think that's what it's for if I interpreted that comment correctly) but not accounted for when checking if the file is too short. So this checks if the header has been artificially extended and doesn't complain about the file being too short if the header was extended beyond the length of the file.